File: JANL/SQL/NonQueryProcedure.vb
Imports System.Data.SqlClient
Imports System.Runtime.CompilerServices
Imports System.Threading.Tasks

Namespace SQL
	''' <summary>
	''' Процедура возвращающая количество задействованных строк
	''' </summary>
	Public Class NonQueryProcedure
		Inherits SQLProcedure(Of Integer)

		''' <summary>
		''' Создаёт новую процедуру именем вызывающего метода
		''' </summary>
		Public Sub New(<CallerMemberName> Optional Name As String = Nothing)
			MyBase.New(Name)
		End Sub

		''' <summary>
		''' Выполнить с указанным соединением
		''' </summary>
		Public Overrides Function Execute(Connection As SqlConnection) As Integer
			SQLCommand.Connection = Connection
			Dim Result = SQLCommand.ExecuteNonQuery()
			Return Result
		End Function

		''' <summary>
		''' Асинхронно выполнить с указанным соединением
		''' </summary>
		Public Overrides Async Function ExecuteAsync(Connection As SqlConnection) As Task(Of Integer)
			SQLCommand.Connection = Connection
			Dim Result = Await SQLCommand.ExecuteNonQueryAsync()
			Return Result
		End Function

	End Class
End Namespace

File: JANL/SQL/NonQueryProcedure.cs
﻿using System.Data.SqlClient;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;

namespace JANL.SQL
{
    /// <summary>
    /// Процедура возвращающая количество задействованных строк
    /// </summary>
    public class NonQueryProcedure : SQLProcedure<int>
    {
        /// <summary>
        /// Создаёт новую процедуру с именем вызывающего метода
        /// </summary>
        public NonQueryProcedure([CallerMemberName] string Name = null) : base(Name) { }

        /// <summary>
        /// Выполнить с указанным соединением
        /// </summary>
        public override int Execute(SqlConnection Connection)
        {
            Command.Connection = Connection;
            var Result = Command.ExecuteNonQuery();
            return Result;
        }

        /// <summary>
        /// Асинхронно выполнить с указанным соединением
        /// </summary>
        public override async Task<int> ExecuteAsync(SqlConnection Connection)
        {
            Command.Connection = Connection;
            var Result = await Command.ExecuteNonQueryAsync();
            return Result;
        }
    }
}